src/http_workers/work_runner.rs

```diff
@@ -131,11 +131,20 @@ where
             if let Ok(bytes) = hyper::body::to_bytes(response.into_body()).await {
                 log::error!("response: {:?}", String::from_utf8(bytes.to_vec()));
             };
-
-            Err(SendError::Terminal(format!(
-                "received error {} from remote twin",
-                status
-            )))
+            if status.is_server_error()
+                && status != http::StatusCode::NOT_IMPLEMENTED
+                && status != http::StatusCode::HTTP_VERSION_NOT_SUPPORTED
```

Comment:
If you wanna handle ALL `500` codes. you instead could do `if (status.code & 500) == 500` this will only be true if and only if the code is `5xx`.

That being said, I don't believe it's a good idea to handle 500 code. 500 codes are not likely to be fixed on retry and I would avoid having special case handling.